File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/domain/entity/OnMessageListener.java
package ru.usedesk.chat_sdk.internal.domain.entity;

import ru.usedesk.chat_sdk.external.entity.UsedeskMessage;
import ru.usedesk.chat_sdk.internal.data.framework.socket.entity.response.Setup;

public interface OnMessageListener {

    void onInit(String token, Setup setup);

    void onInitChat();

    void onNew(UsedeskMessage message);

    void onFeedback();

    void onTokenError();

    void onSetEmailSuccess();
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/domain/entity/OnMessageListener.kt
package ru.usedesk.chat_sdk.internal.domain.entity

import ru.usedesk.chat_sdk.external.entity.UsedeskMessage
import ru.usedesk.chat_sdk.internal.data.framework.socket.entity.response.InitChatResponse

interface OnMessageListener {
    fun onInit(initChatResponse: InitChatResponse)

    fun onInitChat()

    fun onNew(message: UsedeskMessage)

    fun onFeedback()

    fun onTokenError()
}